src/modules/user/user.resolver.ts

```diff
@@ -47,39 +48,32 @@ export class UserResolver {
   }
 
   @Mutation(() => User)
-  async resetPassword(@Arg("input") input: ResetPasswordInput): Promise<User> {
-    const { email, newPassword } = input;
+  async resetPassword(
+    @Arg("resetPasswordInput") input: ResetPasswordInput
+  ): Promise<User> {
+    const updatedUser = await this.authService.resetPassword(input);
 
-    // Check if user exists
-    const user = await prisma.user.findUnique({ where: { email } });
-    if (!user) {
-      throw new Error("User not found");
-    }
-
-    // Generate new password hash
-    const salt = await bcrypt.genSalt(10);
-    const passwordHash = await bcrypt.hash(newPassword, salt);
+    return updatedUser;
+  }
 
-    // Update user's password hash
-    const updatedUser = await prisma.user.update({
-      where: { email },
-      data: { passwordHash },
-    });
+  @Mutation(() => String)
+  async deactivateAccount(@Arg("email") email: string): Promise<string> {
+    const message = await this.userService.deactivateAccount(email);
```

Comment:
Can we rename this variable to something a little more descriptive? Maybe a deactivateAccountMessage or something like that?